pkg/planner/core/point_get_plan.go

```diff
@@ -446,32 +385,32 @@ func (p *PointGetPlan) PrunePartitions(sctx sessionctx.Context) (bool, error) {
 	}
 	row := make([]types.Datum, len(p.TblInfo.Columns))
 	if p.HandleConstant == nil && len(p.IndexValues) > 0 {
-		partColsNames := pt.Meta().Partition.Columns
-		if len(partColsNames) > 0 {
-			partIdx, done, err := needsPartitionPruning(sctx, p.TblInfo, pt, p.dbName, p.IndexInfo, p.IdxCols, p.IndexValues, p.AccessConditions, p.PartitionNames)
-			if table.ErrNoPartitionForGivenValue.Equal(err) {
-				err = nil
-				partIdx = nil
+		hasNonBinaryCollate := false
+		for _, col := range p.IdxCols {
+			switch col.GetType(sctx.GetExprCtx().GetEvalCtx()).GetCollate() {
```

Comment:
We can move this ```sctx.GetExprCtx().GetEvalCtx()```  outside the for-loop.

Reply:
Done